exist-core/src/main/java/org/exist/xquery/value/DoubleValue.java

```diff
@@ -168,78 +167,46 @@ public boolean isPositive() {
     @Override
     public AtomicValue convertTo(final int requiredType) throws XPathException {
         switch (requiredType) {
-            case Type.ATOMIC:
-            case Type.ITEM:
-            case Type.NUMBER:
-            case Type.DOUBLE:
+            case Type.ATOMIC, Type.ITEM, Type.NUMBER, Type.DOUBLE -> {
```

Comment:
I think this `->` syntax is less readable than the previous c-style syntax. The `->` operator here doesn't really deliver anything and should be reserved for the operations it is intended for

Reply:
Although this new style needs getting used to it does improve switch-statements:
- case blocks are enclosed in curly braces and can not fall-through by accident
- fall-through is now explicit, the number of "cases" is clearly visible

How about listing each type in a separate line as it was before? That could improve readability further.

Reply:
It would be interesting to see if that helps.

Reply:
Please have another look, then.

Reply:
@adamretter I would prefer to use of the new switch as function approach instead of the old style here too as it gets more compact.